sense-android-library/src/nl/sense_os/service/phonestate/SensePhoneState.java

```diff
@@ -292,9 +292,40 @@ protected SensePhoneState(Context context) {
         alarmReceiver = new PeriodicPollAlarmReceiver(this);
     }
 
+    private void checkCallState() {
+        int state = telMgr.getCallState();
+
+        JSONObject json = new JSONObject();
+        try {
+            switch (state) {
+            case TelephonyManager.CALL_STATE_IDLE:
+                json.put("state", "idle");
+                break;
+            case TelephonyManager.CALL_STATE_OFFHOOK:
+                json.put("state", "calling");
+                break;
+            case TelephonyManager.CALL_STATE_RINGING:
+                json.put("state", "ringing");
+                break;
+            default:
+                Log.e(TAG, "Unexpected call state: " + state);
+                return;
+            }
+        } catch (JSONException e) {
+            Log.e(TAG, "JSONException in onCallChanged", e);
+            return;
+        }
+
+        Log.v(TAG, "sample call state" + json.toString());
+        // immediately send data point
+        sendDataPoint(SensorNames.CALL_STATE, json.toString(), SenseDataTypes.JSON);
+
+    }
+
     @Override
     public void doSample() {
        //Log.v(TAG, "Do sample");
+        checkCallState();
```

Comment:
@orhan89 yes that's true. But missing an event is probably not due to the PhoneStateListener not receiving it, but that a module is probably not running or has other problems which then doesn't receives this. But I guess it's not a problem to check the current state in the doSample. But I guess next to the call state we need to do this for the other sensors also.